qualtran/bloqs/state_preparation/sparse_state_preparation_via_rotations_test.py

```diff
@@ -0,0 +1,69 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+
+from qualtran import BloqBuilder
+from qualtran.bloqs.basic_gates import IntState
+from qualtran.bloqs.rotations import PhaseGradientState
+from qualtran.bloqs.state_preparation.sparse_state_preparation_via_rotations import (
+    _sparse_state_prep_via_rotations,
+    SparseStatePreparationViaRotations,
+)
+
+
+def test_examples(bloq_autotester):
+    bloq_autotester(_sparse_state_prep_via_rotations)
```

Comment:
this is triggering scipy errors on my machine. Is there a particular version of scipy that is needed?

Reply:
```
bloq_autotester = <function assert_bloq_example_make_for_pytest at 0x11a417e20>

    def test_examples(bloq_autotester):
>       bloq_autotester(_sparse_state_prep_via_rotations)

sparse_state_preparation_via_rotations_test.py:28: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../conftest.py:39: in assert_bloq_example_make_for_pytest
    qlt_testing.assert_bloq_example_make(bloq_ex)
../../testing.py:368: in assert_bloq_example_make
    bloq = bloq_ex.make()
../../_infra/bloq_example.py:53: in make
    return self._func()
sparse_state_preparation_via_rotations.py:184: in _sparse_state_prep_via_rotations
    sparse_state_prep_via_rotations = SparseStatePreparationViaRotations.from_sparse_array(
sparse_state_preparation_via_rotations.py:101: in from_sparse_array
    sparse_coeffs = scipy.sparse.dok_array(coeffs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[TypeError("'NoneType' object is not iterable") raised in repr()] dok_array object at 0x17aab0690>
arg1 = array([0.70914953, 0.        , 0.        , 0.        , 0.46943701,
       0.        , 0.2297245 , 0.        , 0.        , 0.32960471,
       0.        , 0.        , 0.33959273, 0.        , 0.        ])
shape = None, dtype = None, copy = False

    def __init__(self, arg1, shape=None, dtype=None, copy=False):
        _spbase.__init__(self)
        self._dict = {}
    
        self.dtype = getdtype(dtype, default=float)
        if isinstance(arg1, tuple) and isshape(arg1):  # (M,N)
            M, N = arg1
            self._shape = check_shape((M, N))
        elif issparse(arg1):  # Sparse ctor
            if arg1.format == self.format and copy:
                arg1 = arg1.copy()
            else:
                arg1 = arg1.todok()
    
            if dtype is not None:
                arg1 = arg1.astype(dtype, copy=False)
    
            self._dict.update(arg1)
            self._shape = check_shape(arg1.shape)
            self.dtype = arg1.dtype
        else:  # Dense ctor
            try:
                arg1 = np.asarray(arg1)
            except Exception as e:
                raise TypeError('Invalid input format.') from e
    
            if len(arg1.shape) != 2:
>               raise TypeError('Expected rank <=2 dense array or matrix.')
E               TypeError: Expected rank <=2 dense array or matrix.
```

Reply:
I am using the one pinned in the dependencies, 1.14.0. Which version are you using?

Reply:
1.12.0

Reply:
which was released jan, 2024.

We should generally aim to be compatible with each version permissible by `runtime.txt`, so if you need a particularly new feature: provide a minimum version. Unless there's a really really important new feature in the past 8 months of scipy; we should probably re-work the example to be more compatible

Reply:
#1308